nx/lib/nx/defn/compiler.ex

```diff
@@ -48,7 +48,7 @@ defmodule Nx.Defn.Compiler do
 
   # These operations do not have valid meaning for Nx.Defn.Expr
   @forbidden_ops [:backend_copy, :backend_deallocate, :backend_transfer] ++
-                   [:to_binary, :to_scalar, :to_flat_list, :to_heatmap, :to_batched_list] ++
+                   [:to_binary, :to_number, :to_scalar, :to_flat_list, :to_heatmap, :to_batched_list] ++
```

Comment:
This is annoying the formatter and without this the code will fail to compile anyway, so we are fine.

```suggestion
                   [:to_binary, :to_number, :to_flat_list, :to_heatmap, :to_batched_list] ++
```